### PR TITLE
Feature/handle interrupted recog

### PR DESCRIPTION
**Pull request by runningferret**
Handle an interrupted UniMRCP request. Since the AMI Event for Hangup comes through separately we should just return a NoMatch for the recognition portion and let the rest of Punchblock deal with the Hangup event.

### CODE REVIEW

**Comment by runningferret** [comment]
@sfgeorge @jared-prime

File: lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb [hunk 1]
```diff
@@ -39,7 +39,6 @@ def validate
 
           def execute_app(app, *args)
             UniMRCPApp.new(app, *args, unimrcp_app_options).execute @call
-            raise UniMRCPError if @call.channel_var('RECOG_STATUS') == 'ERROR'
```
Comment (Jared-Prime) [inline lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb:42]: :cat:

File: lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb [hunk 2]
```diff
@@ -87,6 +86,14 @@ def input_node
           end
 
           def complete
+            recog_status = @call.channel_var('RECOG_STATUS')
```
Comment (Jared-Prime) [inline lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb:89]: quantum entangled :cat:

File: lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb [hunk 3]
```diff
@@ -87,6 +86,14 @@ def input_node
...
+
+            raise UniMRCPError if recog_status == 'ERROR'
+
+            if recog_status == 'INTERRUPTED'
+              return Punchblock::Component::Input::Complete::NoMatch.new
```
Comment (Jared-Prime) [inline lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb:94]: this does the trick :cat:

**Comment by Jared-Prime** [comment]
looks good to me :+1:

**Comment by ajayanty** [comment]
Waiting for @sfgeorge final review

File: lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb (hunk 3 above)
Comment (sfgeorge) [inline lib/punchblock/translator/asterisk/component/mrcp_recog_prompt.rb:94]: Thank you for investigating this.  Not easy, but very nicely done!

**Comment by sfgeorge** [comment]
Code looks great.  Please add something to CHANGELOG.md and we'll be set to go!

**Comment by runningferret** [comment]
@sfgeorge @Jared-Prime

**Comment by Jared-Prime** [comment]
:+1:

**Comment by sfgeorge** [comment]
Looks great, thank you! :+1: 

@ajayanty Can you merge this for us?

**Comment by ajayanty** [comment]
This has been merged into cloudvox:develop-ifbyphone

**Pull request merged**